docs/reference/setup/upgrade.asciidoc

```diff
@@ -7,6 +7,7 @@ Before upgrading Elasticsearch:
 
 * Consult the <<breaking-changes,breaking changes>> docs.
 * Test upgrades in a dev environment before upgrading your production cluster.
+* Remove your current plugins, then install the latest after upgrading.
```

Comment:
I think instead of a blanket "Remove and then reinstall" perhaps a "You should ensure you upgrade or remove non-compatible plugins when upgrading Elasticsearch" would work, what do you think?